src/sources/dune.py

```diff
@@ -2,34 +2,57 @@
 Source logic for Dune Analytics.
 """
 
+import re
 from abc import ABC
-from typing import Type, Any, Literal
+from typing import Type, Any, Literal, Union
 
 import pandas as pd
 from dune_client.client import DuneClient
 from dune_client.models import ExecutionResult
 from dune_client.query import QueryBase
 from pandas import DataFrame
 from sqlalchemy import BIGINT, BOOLEAN, VARCHAR, DATE, TIMESTAMP
-from sqlalchemy.dialects.postgresql import BYTEA, DOUBLE_PRECISION, INTEGER
+from sqlalchemy.dialects.postgresql import BYTEA, DOUBLE_PRECISION, INTEGER, NUMERIC
 
 from src.interfaces import Source, TypedDataFrame
+from src.logger import log
 
-DUNE_TO_PG: dict[str, Type[Any]] = {
+DUNE_TO_PG: dict[str, Union[Type[Any], NUMERIC]] = {
```

Comment:
I do not think that this is the correct way to coerce the typing here, but I couldn't figure out anything else.

The root of the problem is that we need to map to an instance of `NUMERIC` to preserve the Dune-specified precision and width. Mapping to simply `NUMERIC` (as a class, not an instance) would lose digits past the decimal, for example the value `12.001` that we're testing with would end up stored as `12`.
Open to suggestions if you think it can be handled differently.

Reply:
I will have to checkout this branch to see what's going on here. 

Nit: `Union[X, Y]` is the same as `X | Y` since ~ 3.8